unified-codes/apps/web/src/components/containers/detail/DetailPage.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import DetailAttributeList from './DetailAttributeList';
+import DetailChildList from './DetailChildList';
+import DetailPropertyList from './DetailPropertyList';
+import DetailLayout from '../../layout/DetailLayout';
+
+import { IDetailAction } from '../../../actions';
+import { IState } from '../../../types';
+
+export interface DetailPageProps {
+    onMount: () => void;
+    onUnmount: () => void;
+}
+
+export type DetailPage = React.FunctionComponent<DetailPageProps>;
+
+export const DetailPageComponent: DetailPage = ({ onMount, onUnmount }) => {
+    React.useEffect(() => {
+        onMount();
+        return onUnmount;
+    }, []);
```

Comment:
Proposed convention: can only navigate to page components + all pages use HOCs and have `onMount`, `onUnmount` effects. Suggest this is a decent convention for handling navigation (e.g. if `DetailsPage` navigated to via URL not browser).